### PR TITLE
lnd, lightning-loop: add certificate options `extraIPs` and `extraDomains`

### PR DESCRIPTION
**Pull request by erikarvstedt**
#### Copy of commit msg
This is useful for non-local access to the lnd REST server.

@sputn1ck, does this fit your use case?

### CODE REVIEW

**Review by sputn1ck — DISMISSED** [review]
Cool PR, exactly what I needed. Now I just need pool and loop to be able to listen to multiple rpc hosts and my remote LIT setup is perfect (currently using ssh tunneling to forward the grpc  connections).

File: modules/lnd.nix [hunk 1]
```diff
@@ -276,9 +302,14 @@ in {
...
-    nix-bitcoin.generateSecretsCmds.lnd = ''
+    nix-bitcoin.generateSecretsCmds.lnd = let
+      extraAltNames = builtins.concatStringsSep "," (
+        (map (ip: "IP:${ip}") cfg.certificate.extraIPs) ++
+        (map (domain: "DNS:${domain}") cfg.certificate.extraDomains)
+      );
+    in ''
       makePasswordSecret lnd-wallet-password
-      makeCert lnd '${optionalString (cfg.rpcAddress != "localhost") "IP:${cfg.rpcAddress}"}'
+      makeCert lnd '${extraAltNames}"}'
```
Comment (sputn1ck) [inline modules/lnd.nix:312]: I think `localhost` should always be added, as lncli/loop/pool is broken if not.

**Comment by erikarvstedt** [comment]
> be able to listen to multiple rpc hosts 

You can use `extraConfig`:
```nix
services.lightning-loop.extraConfig = ''
   rpclisten=myhost:myport
'';
```

**Comment by erikarvstedt** [comment]
I've now added the same options to `lightning-loop`, which uses the same certificate setup mechanism as `lnd`.

**Comment by sputn1ck** [comment]
> > be able to listen to multiple rpc hosts
> 
> You can use `extraConfig`:
> 
> ```nix
> services.lightning-loop.extraConfig = ''
>    rpclisten=myhost:myport
> '';
> ```

I'm pretty sure loop and pool currently can only listen on one rpc host, so it would just take the last one you define in `services.lightning-loop.extraConfig`

**Comment by erikarvstedt** [comment]
Ah, I thought it worked like in `lnd`.
Until this is fixed upstream, you could use `socat` or `nginx` for TCP forwarding.

File: modules/lightning-loop.nix [hunk 2]
```diff
@@ -40,13 +40,37 @@ let
...
       type = types.lines;
       default = "";
       example = ''
         debuglevel=trace
       '';
-      description = "Extra lines appended to the configuration file.";
+      description = ''
+        Extra lines appended to the configuration file.
+        See here for all available options:
+        https://github.com/lightninglabs/loop/blob/11ab596080e9d36f1df43edbeba0702b25aa7457/loopd/config.go#L119
```
Comment (nixbitcoin) [inline modules/lightning-loop.nix:72]: This should point to master

**Review by jonasnick — APPROVED** [review]
ACK c853f380d43279813a8c4ae2a955f9cbf0bb7231

**Pull request merged**